src/app/components/add-card/add-card.component.ts

```diff
@@ -0,0 +1,75 @@
+import {
+  Component,
+  OnInit,
+  ElementRef,
+  ViewChild,
+  Input,
+  Output,
+  EventEmitter,
+} from '@angular/core';
+import { formatDate } from '@angular/common';
+import * as DOMPurify from 'dompurify';
+
+import { UtilityService } from 'src/app/utility.service';
+import { TaskService } from 'src/app/task.service';
+import { ConstantsService } from 'src/app/constants.service';
+
+@Component({
+  selector: 'app-add-card',
+  templateUrl: './add-card.component.html',
+  styleUrls: ['./add-card.component.css'],
+})
+export class AddCardComponent implements OnInit {
+  constructor(
+    public utilityService: UtilityService,
+    public taskService: TaskService,
+    public constantsService: ConstantsService
+  ) {}
+
+  ngOnInit(): void {}
+
+  @ViewChild('newTaskInput') newTaskInput!: ElementRef;
+  name: string = '';
+  id: number = 0;
+  done: boolean = false;
+  edit: boolean = false;
+  trash: boolean = false;
+  lastId: number = 0;
+  startDate: number = Date.now();
```

Comment:
Instead of list of variables, I would move them to a model and created Task type.

Reply:
done!